assets/js/cart.js

```diff
@@ -66,11 +66,17 @@ storefrontApp.controller('cartController', ['$rootScope', '$scope', '$timeout',
         if ($scope.formCart.$invalid) {
             return;
         }
-        if ($scope.cart.hasPhysicalProducts) {
-            $scope.outerRedirect($scope.baseUrl + 'cart/checkout');
-        } else {
-            $scope.outerRedirect($scope.baseUrl + 'cart/checkout');
+        if ($scope.cart.comment) {
+            cartService.updateCartComment($scope.cart.comment).then(function (resp) {
+            });
         }
+        $timeout(function () { 
```

Comment:
That such code should not be at all!!! 
You should chain the $promise operation of update command with redirection operation